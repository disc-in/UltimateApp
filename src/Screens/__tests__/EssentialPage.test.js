import React from 'react';
import { render } from '@testing-library/react-native';
import { Provider } from 'react-redux';

import store from '../../Store/testStore';

import ConectedEssentialPage from '../EssentialPage';

describe('<EssentialPage />', () => {
  it('renders correctly', () => {
    const { toJSON } = render(
      <Provider store={store}>
        <ConectedEssentialPage />
      </Provider>,
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
