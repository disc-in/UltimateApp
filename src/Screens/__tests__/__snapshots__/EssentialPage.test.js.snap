// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EssentialPage /> renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "height": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomColor": "#e5e5e5",
        "borderBottomWidth": 2,
        "flexDirection": "row",
        "height": 70,
        "padding": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      Choose a topic:
    </Text>
    <View
      style={
        Object {
          "flex": 1,
          "marginHorizontal": 20,
        }
      }
    >
      <RNCPicker
        items={
          Array [
            Object {
              "label": "Cutting",
              "testID": undefined,
              "textColor": undefined,
              "value": 0,
            },
            Object {
              "label": "Throwing",
              "testID": undefined,
              "textColor": undefined,
              "value": 1,
            },
          ]
        }
        numberOfLines={1}
        onChange={[Function]}
        selectedIndex={1}
        style={
          Array [
            Object {
              "height": 216,
            },
            Object {
              "fontSize": 14,
            },
          ]
        }
      />
    </View>
  </View>
  <RCTScrollView
    data={
      Array [
        Object {
          "id": 1,
          "text": "Rise Up Ultimate",
          "title": "How to make a perfect throw?",
          "video": "424000350",
          "youtube": "https://youtu.be/CqpPfVZ1HeI",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "backgroundColor": "#fff",
      }
    }
    viewabilityConfigCallbackPairs={Array []}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "borderBottomColor": "#e5e5e5",
                "borderBottomWidth": 1,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 16,
                  "fontWeight": "bold",
                  "paddingLeft": 10,
                  "paddingVertical": 15,
                }
              }
            >
              How to make a perfect throw?
            </Text>
            <View
              accessibilityIgnoresInvertColors={true}
              style={
                Object {
                  "height": 250,
                }
              }
            >
              <Image
                source={
                  Object {
                    "uri": undefined,
                  }
                }
                style={
                  Array [
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    },
                    Object {
                      "height": 250,
                      "width": undefined,
                    },
                    undefined,
                  ]
                }
              />
              <View
                style={
                  Object {
                    "backgroundColor": "#232b2b",
                    "bottom": 10,
                    "paddingHorizontal": 5,
                    "position": "absolute",
                    "right": 5,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#fff",
                      "fontSize": 16,
                    }
                  }
                />
              </View>
            </View>
            <Text
              style={
                Object {
                  "fontSize": 14,
                  "paddingLeft": 10,
                  "paddingVertical": 15,
                  "textAlign": "center",
                }
              }
            >
              Rise Up Ultimate
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
